presto-base-jdbc/src/main/java/io/prestosql/plugin/jdbc/BaseJdbcClient.java

```diff
@@ -305,7 +306,166 @@ protected static ResultSet getColumns(JdbcTableHandle tableHandle, DatabaseMetaD
+            DatabaseMetaData metaData = connection.getMetaData();
+            // determine a column to split the whole table scan
+            ColumnInfo splitColumn = getSplitColumn(metaData, tableHandle.getCatalogName(), tableHandle.getSchemaName(), tableHandle.getTableName());
+
+            if (splitColumn != null) {
+                log.info("Table: %s.%s.%s, splitColumn: %s", tableHandle.getCatalogName(), tableHandle.getSchemaName(), tableHandle.getTableName(), splitColumn);
+
+                // collect the min/max stats for the specified split column
+                String minMaxSql = format("select min(%s) AS MIN_VALUE, max(%s) AS MAX_VALUE from %s",
+                        splitColumn.getName(), splitColumn.getName(),
+                        quoted(tableHandle.getCatalogName(), tableHandle.getSchemaName(), tableHandle.getTableName()));
+                try (Statement statement = connection.createStatement()) {
+                    log.info("Execute sql: %s", minMaxSql);
+
+                    statement.execute(minMaxSql);
+                    try (ResultSet rs = statement.getResultSet()) {
+                        rs.next();
+                        long minId = rs.getLong("MIN_VALUE");
+                        long maxId = rs.getLong("MAX_VALUE");
+                        int splitCount = deducePartitionCount(maxId - minId);
+
+                        if (splitCount > 1) {
+                            List<Pair<Long, Long>> partitions = partition(minId, maxId, splitCount);
+                            ranges = new ArrayList<>();
+                            for (Pair<Long, Long> x : partitions) {
+                                ranges.add(new JdbcSplit.RangeInfo(splitColumn.getName(), x.getKey(), x.getValue()));
+                            }
+
+                            // add a null range if the column is nullable
+                            if (splitColumn.isNullable()) {
+                                ranges.add(new JdbcSplit.RangeInfo(splitColumn.getName(), null, null));
+                            }
+                        }
+                    }
+                }
+            }
+        }
+        catch (SQLException e) {
+            log.error("getSplits failed!", e);
+        }
+
+        List<JdbcSplit> splits = new ArrayList<>();
+        if (ranges != null) {
+            for (JdbcSplit.RangeInfo range : ranges) {
+                splits.add(new JdbcSplit(Optional.empty(), Optional.of(range)));
+            }
+        }
+        else { // no split column
+            JdbcSplit jdbcSplit = new JdbcSplit(Optional.empty(), Optional.empty());
+            splits.add(jdbcSplit);
+        }
+
+        return new FixedSplitSource(splits);
+    }
+
+    /**
+     * Looking for the split column based on some predefined rules:
+     * 1. If there is a auto_increment column, pick it.
+     * 2. Otherwise if there is a bigint column, pick it.
+     * 3. Otherwise if there is an integer column, pick it.
```

Comment:
This is very arbitrary and doesn't use the underlying organization of the data.
We may end up using running several/many splits and each of them will do full table scan (or worse) in the RDBMS.

What I would prefer to see here is finding a _partitioning_ column.
This will allow splits to read from separate partitions, reading non-overlapping data sets and thus not increasing load on the RDBMS.

Of course, this is not possible to do in generic way, so I think this needs to be addressed on connector level, not in the generic base class.
For example, you may want to start with PostgreSQL.